addons/pos_loyalty/models/pos_order.py

```diff
@@ -74,6 +74,7 @@ def confirm_coupon_programs(self, coupon_data):
             'partner_id': p.get('partner_id', False),
             'code': p.get('barcode') or self.env['loyalty.card']._generate_code(),
             'points': 0,
+            'expiration_date': p['date_to'],
             'source_pos_order_id': self.id,
```

Comment:
very small issue, but for people that would update sources without restarting odoo (which should not happen on saas and odoo.sh I think), shouldn't we allow `date_to` to be missing, eg:

```suggestion
            'source_pos_order_id': self.id,
            **{'expiration_date': p['date_to']} if p.get('date_to') else {},
```
(or just `p.get('date_to')` if it was ok to give `None` as `expiration_date` (there is no default but a customization could have it))